cmd/icinga-notifications/main.go

```diff
@@ -70,8 +70,19 @@ func main() {
 	defer cancel()
 
 	logger.Infof("Connecting to database at '%s'", db.GetAddr())
-	if err := db.PingContext(ctx); err != nil {
-		logger.Fatalf("Cannot connect to the database: %+v", err)
+	{
+		// When installing Icinga Notifications via our packages, the systemd service will be started automatically
+		// e.g. on Debian based systems, and since the unit type is set to `notify`, make sure you don't block the
+		// installation if no database is configured yet.
+		dbCtx, cancelFunc := context.WithTimeout(ctx, 5*time.Second)
+		defer cancelFunc()
```

Comment:
This `defer` seems useless. In case of a `logger.Fatalf`, the logger exits the program with a call to `os.Exit` and otherwise, `cancelFunc` will be called directly.